### ISSUE TITLE
Feature request: Form view

### ISSUE THREAD

**Issue by M-Rick** [sqlitebrowser/sqlitebrowser#127]
It would be great to a get a form view in the way of Bento or Glom. It would be more attractive and more convenient to enter datas.

**Comment by M-Rick** [comment]
Hi

I come back to you with a long time later after the request. I am very sorry, I have been busy and forgot to look after this.

So, to remind the request was to create a simple form view for SQLite browser.

I spoke a bout Glom (Gnome) and Bento (Mac OS X).

They are working almost the same way, but with a simpler way for Bento with a little more graphic features, so I made screenshots of it with Skitch that let me comment them.
Bento has been discontinued, and Mac users are angry about this FileMaker decision, it was also an SQLite based database set, so a Bento database could maybe even imported in SQLite browser.
It could so fill the need, and also give a mire powerful free product available as a multi-plateforme On Linux, Mac and Windows. Glom for Windows is outdated, the Mac version needs to compiled with Mac Ports, it is unstable and not easy at all …
SQLite browser comes there has a better simple product and powerful anyway. It has a nice GUI, and it is very stable and easy to use or to install. So it could be the way to make everybody happy.

The idea is to have a simple view form to manage the datas and having a GUI which is simple and pleasant to enter the datas for the end user.
It should be possible to organize easily the way the fields are displayed and which ones could be visible.
It should be able to manage multiple forms.
It shouldn't let you manage complex features.


you can contact me for questions.

There are available as a download in here, the link will remain available for 15 days from now.
http://freebox-aymeric.no-ip.org/share/tFVcEKqNG4GqE00e/


Best regards

M-Rick

**Comment by justinclift** [comment]
Thanks, downloading that now.  Is it ok if I insert those pictures into this Issue, so others can see them?

**Comment by M-Rick** [comment]
ho yes that's ok

I shared them this way cos' of the heavy file, but now they are yours 😉

**Comment by justinclift** [comment]
Cool.  It looks like you've spent a lot of time and effort thinking this through, so I hope someone picks up this Issue + implements it. :grinning:

![plein_ecran_08_02_15_14_00](https://cloud.githubusercontent.com/assets/406299/6099064/d331a48e-afe6-11e4-987d-465ec33871ff.png)
![plein_ecran_08_02_15_14_10](https://cloud.githubusercontent.com/assets/406299/6099068/d3390e54-afe6-11e4-93fe-fd1655782cec.png)
![plein_ecran_08_02_15_14_24](https://cloud.githubusercontent.com/assets/406299/6099066/d3384866-afe6-11e4-967d-39061840faf0.png)
![plein_ecran_08_02_15_14_56](https://cloud.githubusercontent.com/assets/406299/6099067/d338c82c-afe6-11e4-8522-cb5589896e1a.png)

We already have this one as part of our normal UI:

![plein_ecran_08_02_15_16_16](https://cloud.githubusercontent.com/assets/406299/6099069/d33ad8ce-afe6-11e4-8368-94de5015ed06.png)

![plein_ecran_08_02_15_16_17](https://cloud.githubusercontent.com/assets/406299/6099065/d33741b4-afe6-11e4-8c96-069406e78520.png)
![plein_ecran_08_02_15_18_03](https://cloud.githubusercontent.com/assets/406299/6099070/d3458490-afe6-11e4-8569-06820eb25e06.png)

This bit is making sense to me now for "form views":

![plein_ecran_08_02_15_18_04](https://cloud.githubusercontent.com/assets/406299/6099073/d34db516-afe6-11e4-9aa5-d92b063d7345.png)

So yeah, I get what you're meaning. :smile:

![plein_ecran_08_02_15_18_06](https://cloud.githubusercontent.com/assets/406299/6099071/d34cd22c-afe6-11e4-93ba-6dbac8acf050.png)
![capture_d ecran_2015-02-08_a_18_13_05](https://cloud.githubusercontent.com/assets/406299/6099072/d34d7eac-afe6-11e4-865f-10b87b43b84f.png)
![plein_ecran_08_02_15_18_43](https://cloud.githubusercontent.com/assets/406299/6099074/d34eb4de-afe6-11e4-83c2-135e7286ae07.png)
![plein_ecran_08_02_15_18_46](https://cloud.githubusercontent.com/assets/406299/6099075/d3518718-afe6-11e4-8b35-b3b7828937a4.png)
![plein_ecran_08_02_15_18_54](https://cloud.githubusercontent.com/assets/406299/6099076/d357c182-afe6-11e4-8731-ed1244c0426b.png)
![plein_ecran_08_02_15_18_56](https://cloud.githubusercontent.com/assets/406299/6099077/d3628216-afe6-11e4-9a39-40af17fd9a9c.png)

**Comment by justinclift** [comment]
I guess there would need to be a "form designer" part, and a "form viewer" part (eg for displaying data and/or data entry).

**Comment by M-Rick** [comment]
yes of course, that'd be more convenient.

**Comment by M-Rick** [comment]
yes it took me some time, but because I think this little piece of software is good enough to fill the gap.

but I am also aware that'll be a lot of work for the dev(s) behind… and unfortunately, I am unable to help at this level 😞 
but I only based my view with a discontinued software. Glom for Gnome is interesting too, but it's less handy, but it has for it that it can manage SQLite databases, but PostgreSQL or MySQL databases as well.
I tried LibreOffice/OOo but the form designer is buggy and not convenient at all to be used.

I think the most important functions in this is a form designer/view before the design.

**Comment by M-Rick** [comment]
This database can be a good example as well for a data form GUI.
Records http://pushpopcorn.com

![](http://static1.squarespace.com/static/5052391a24ac3b03d5339e83/54c624ffe4b0f7f4ab66fb1e/54c67deae4b0c0cee6211603/1422294523506/Records+Screen+Shot+1.png)

**Comment by M-Rick** [comment]
Symphytum does this. The GUI is not very beautiful, but the idea is there.

[https://github.com/giowck/symphytum](https://github.com/giowck/symphytum)

**Comment by caesar70** [comment]
Will the form editor ever be implemented  ?

**Comment by chrisjlocke** [comment]
Not in the short term.  However, as this is an open-source project, anyone is welcome to come along and submit code, so maybe someone is bored enough to do just that.  But otherwise, the current team are beavering away on other issues.

**Comment by fadi123go** [comment]
User defined forms are difficult to implement . I am only asking for a parallel  form view for each table similar to symphytum, sqlitestudio and pretty much every rdmbs gui tool.
